astropy/utils/decorators.py

```diff
@@ -203,7 +224,7 @@ def deprecate(
             else:
                 message = (
                     "The {func} {obj_type} is deprecated and may "
-                    "be removed in a future version."
+                    "be removed in a future version (minimum version {removal_version})."
```

Comment:
I would not mix this into this PR as this breaks past behaviour when/if anyone matches with the full message. And I would rather see this feature backported, so more versions can use it.

Reply:
That is fair, I will remove this change.